CODE_OF_CONDUCT.md

```diff
@@ -111,11 +112,11 @@ Community leaders will follow these Community Impact Guidelines in determining t
 
 **Community Impact**: Demonstrating a pattern of violation of community standards, including sustained inappropriate behavior,  harassment of an individual, or aggression toward or disparagement of classes of individuals.
 
-**Consequence**: A permanent ban from any sort of public interaction within the project community.
+**Consequence**: A permanent ban from any sort of public interaction within the community.
```

Comment:
I believe that this would be a highly contentious change.

The concern, expressed by a number of maintainers in the past, is that a project could be forced by the Foundation or the CPC, to ban a contributor for their actions in a different project, without having had any transparency or say in the CoC enforcement process.

I suggest leaving this change out of this particular PR (as this would be reverting an edit made to 2.0, not updating 2.0 to 2.1) and discussing the consistency issue as part of #852.

Reply:
That's not what this means though. "community" doesn't mean it *must* be the entire community, it just means it *may* be. With "project", it *must not* be.

Reply:
IMO `community` reads as _must_ and `project community` reads as _must not_.

Reply:
hmm - to me, it would require a qualifier for that, because "community" is a vague, undefined term.

What about:
```suggestion
**Consequence**: A permanent ban from any sort of public interaction within the community (the project, and potentially more widely).
```

Reply:
I don't think this is the point in the doc where we should be clarifying the term "community". It's also referred to elsewhere, e.g. in the "Temporary Ban" section:
> Consequence: A temporary ban from any sort of interaction or public communication with the community for a specified period of time.

To be clear, that bit is already now in our CoC.

I think the issue here is that as with our CLA, we're using an upstream text that presumes a single "project" or a single "community" as the user; we instead have the wider OpenJS Foundation "project"/"community" as an umbrella for individual named projects.

So really this upstream change isn't the issue; the issue is that we do not include a "community" definition in e.g. a preamble to this document. And that's a problem that's independent of the Contributor Covenant version we're using.

Reply:
> So really this upstream change isn't the issue;

To be clear, this is _not_ an upstream change. There is no modification of that line between 2.0 and 2.1.

Both 2.1 and [2.0](https://github.com/EthicalSource/contributor_covenant/blob/release/content/version/2/0/code_of_conduct.md#4-permanent-ban) read:

> Consequence: A permanent ban from any sort of public interaction within the community.

What's in our CoC is the result of a custom edit of version 2.0.

So although I agree that this edit isn't the best solution, I maintain that it shouldn't be removed as part of updating to 2.1.

Reply:
> What's in our CoC is the result of a custom edit of version 2.0.

This is not the case. There is more than one "version 2.0". See https://github.com/openjs-foundation/cross-project-council/issues/852#issuecomment-1085759325

Reply:
> This is not the case. There is more than one "version 2.0". See https://github.com/openjs-foundation/cross-project-council/issues/852#issuecomment-1085759325

Oof! I hadn't properly registered this. Apologies and thanks for catching it!

Unfortunately, I don't think that changes the fact that some projects will have an issue with this edit.

---

I think our reasonable options are:

1. Don't upgrade for now (and allow JSON Schema to run Contributor Covenant 2.1 until we sort this out), <u>or</u>
2. Update to Contributor Covenant 2.1, but keep permanent bans scoped to the project's community, <u>or</u>
3. First agree on a definition of the meaning and scope of community in #852 that allows us to update to the latest version of Contributor Covenant 2.0 (https://github.com/EthicalSource/contributor_covenant/commit/a63f697fae93de5abe6354af2c4afa78c5804298), and then update to Contributor Covenant 2.1.

Reply:
I don't think options 1 and 2 are viable, given that this discussion has identified the need to define "community" in this context. And as I point out above, this also matters for temporary bans, which even now target the whole "community", without a "project" qualifier.

My sense of prior discussions on this has been that we consider "community" hear to mean the project's community. But let's talk about this at the CPC meeting tomorrow?

Reply:
Was there a resolution at the meeting?

Reply:
Sorry, scratch that, I see @jorydotcom's explanation.  Seems like this conversation could be marked resolved.